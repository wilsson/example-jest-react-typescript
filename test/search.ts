
const emojiMap = [
    { title: 'Frozen', emoji: '❄ �'},
    { title: 'Finding Nemo', emoji: '��'},
    { title: 'Kung Fu Panda', emoji: '���'},
    { title: 'Harry Potter', emoji: '��⚡'},
    { title: 'Hunger Games', emoji: '���'},
    { title: 'The Lion King', emoji: '��'},
    { title: 'ET', emoji: '���'},
];

function searchMovies(query) {
    return emojiMap.filter((movie: any) => {
        return movie.title.includes(query)
    });
}

expect.addSnapshotSerializer({
    test: (val) => val.title && val.emoji,
    print: (val) => `${val.emoji} ${val.title}`
});

it('Search Movies', () => {
    expect(searchMovies('F')).toMatchSnapshot();
})