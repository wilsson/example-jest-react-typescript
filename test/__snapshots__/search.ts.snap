// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Movies 1`] = `
Array [
  ❄ � Frozen,
  �� Finding Nemo,
  ��� Kung Fu Panda,
]
`;
